### DISCUSSION TITLE
Enable loading templates from private repositories

### DISCUSSION THREAD

**Discussion by forgedhallpass** [projectdiscovery/nuclei#3431] [Q&A] [answered]
Similarly to [URL based template input support ](#970) and [Running remote hosted template](https://github.com/projectdiscovery/nuclei/issues/1216), but also support running templates stored on remote private repositories (e.g. GitHub).

This would be helpful for companies that store their templates on version control and want to execute them against their internal targets for the purpose of security regression testing.

Loading them as direct reference, folder or the whole repository would enable the most flexibility.

Possible solution would be do download the whole folder or repository if the input is not a direct link to a template and then run it from the disk, otherwise the references e.g. within a workflow would need to be made relative to the repository URL.

**Comment by ehsandeep** [comment, marked as answer]
@forgedhallpass, this should be supported already; please find the feature details in the issue https://github.com/projectdiscovery/nuclei/issues/2613.

Created follow up ticket for updating the docs - https://github.com/projectdiscovery/nuclei-docs/issues/137

**Comment by forgedhallpass** [comment]
Great, thank you @ehsandeep!
